apps/consumer-client/src/pages/examples/third-party-auth/utils.ts

```diff
@@ -0,0 +1,154 @@
+import { EdDSATicketPCDPackage } from "@pcd/eddsa-ticket-pcd";
+import {
+  constructZupassPcdGetRequestUrl,
+  getWithoutProvingUrl,
+  openZupassPopup
+} from "@pcd/passport-interface";
+import { ArgumentTypeName } from "@pcd/pcd-types";
+import { SemaphoreIdentityPCDPackage } from "@pcd/semaphore-identity-pcd";
+import { SemaphoreSignaturePCDPackage } from "@pcd/semaphore-signature-pcd";
+import {
+  EdDSATicketFieldsToReveal,
+  ZKEdDSAEventTicketPCDArgs,
+  ZKEdDSAEventTicketPCDPackage
+} from "@pcd/zk-eddsa-event-ticket-pcd";
+import { CONSUMER_SERVER_URL, ZUPASS_URL } from "../../../constants";
+
+/**
+ * Requests a one-time challenge (nonce) for authentication with an EdDSA ticket PCD 
+ * from the `consumer-server`.
+ */
+export async function generateNonce(): Promise<string> {
+  const response = await fetch(`${CONSUMER_SERVER_URL}auth/nonce`, {
```

Comment:
It would be better to use the `url-join` package to join URL strings. We have had some problems in `passport-client` due to concatenating URL strings, and `url-join` provides a much safer solution.